### ISSUE TITLE
Question: How to style a graph only when it's empty?

### ISSUE THREAD

**Issue by fearphage** [plotly/plotly.js#1263]
When the graph has data, all is well. When the graph is empty, I get useless values shown. Is there a way to manually force the layout when the graph is empty, but allow it to automatically be managed when there is data? 

I could do this manually by checking the data that gets passed to Plotly and adjusting the layout myself. Is there a better way?

**Comment by etpinard** [comment]
> When the graph is empty, I get useless values shown

As per the contributing guidelines (that you just deleted), can you please provide a reproducible example?

**Comment by fearphage** [comment]
Absoutely https://jsfiddle.net/rzz541km/2/

**Comment by etpinard** [comment]
You mean something like this https://jsfiddle.net/rzz541km/3/ ?

**Comment by fearphage** [comment]
Yes precisely that, but I only want to do that when there is nothing to plot. Plotly does the right thing when there is data present. Is it possible to programmaticalyl make that distinction via settings? or do I need to check `data.length === 0` and do some patching on my end?

**Comment by etpinard** [comment]
> but I only want to do that when there is nothing to plot.

I don't understand. What's the problem in my fiddle in https://github.com/plotly/plotly.js/issues/1263#issuecomment-268549831

**Comment by fearphage** [comment]
> What's the problem in my fiddle

Compare your graph (https://jsfiddle.net/rzz541km/3/) to mine (https://jsfiddle.net/rzz541km/4/). My only change was to add data to the graph. When you double click my graph now, you get closer to what I would expect to be presented. 

Let's say we're graphing the number of github tickets I open in a day. When the graph is empty, it could potentially show -1 - 2 on the y axis and -1 to 5 on x axis. In the context of what I'm graphing, negative isn't meaningful. I can't open negative tickets.

My ultimate goal is sane graph defaults I believe. I want to anchor the graph with meaningful numbers when it's empty, but when there is data, I want it to graph things like normal and essentially ignore the anchoring. Without the anchoring, I get graphs with negative numbers which don't make sense in the context of my graph.

Do you understand where I'm coming from? Does my request make sense?

**Comment by etpinard** [comment]
I still don't get it.

Maybe if you could tell _how_ you add data to your graph would help. Are you using `Plotly.addTraces`?
